### DISCUSSION TITLE
RustDesk Server key mismatch

### DISCUSSION THREAD

**Discussion by ultradj83** [rustdesk/rustdesk#7012] [Q&A]
Dears,
just installed RustDesk Server from MyQnap store.

I copied the public key from /opt/RustDeskServer/id_ed25519.pub to clients but, started any connections, I'm returned key mismatch.

RustDesk-tools doctor reports all fine (but the API port I won't use).

Any suggestions?

Regards

**Comment by jack7803m** [comment]
Currently having the exact same issue. Just switched to self-hosted due to the public servers having trouble today and but I can't seem to get it to work - clients always give a "key mismatch" error

**Reply by ultradj83** [reply]
Follow this instead: https://www.youtube.com/watch?v=PjY5TpWMNz4

**Reply by jack7803m** [reply]
Unfortunately I'm not using that system of install, I just ran the automated install script off of https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/install/ 
I tried changing the hbbs and hbbr program arguments to the same as shown in the video, but still getting a key mismatch error.

**Reply by ultradj83** [reply]
This is specific for Qnap but could be extended to any devices supporting docker.

**Reply by jack7803m** [reply]
Ended up resolving the issue by just removing the arguments from `hbbs` and not entering any key on the clients. This is a temporary fix, as I would like to use a key in the future, but I guess it's fine for now. 

I may try resetting the server and installing with docker instead later.

**Comment by marcoczen** [comment]
> Ended up resolving the issue by just removing the arguments from `hbbs` and not entering any key on the clients. This is a temporary fix, as I would like to use a key in the future, but I guess it's fine for now.
> 

Hi. Did u get this solved ? Mind sharing ?

**Reply by ultradj83** [reply]
Buggy build.
I worked around by installing the docker version.

**Comment by arodriguezd** [comment]
In my case when the private/public keys were created. The private key did not have the correct permissions. It was the same as the public key

```
-rw-r--r-- 1 root root    88 Apr  31 10:02 id_ed25519
-rw-r--r-- 1 root root    44 Apr  31 10:02 id_ed25519.pub
```
Change the permissions of the private key and now they are, also restart the dockers and I started to work. 
```
-rw------- 1 root root    88 Apr  31 10:02 id_ed25519
-rw-r--r-- 1 root root    44 Apr  31 10:02 id_ed25519.pub
```

**Reply by cyberphen** [reply]
Thanx this fixed the issue. I am using a docker version, had id_ed25519 and id_ed25519.pub in ./data folder inside the container which was also mounted to the host. Changing the permissions as this comment and copying the key from .pub file to the clients works.

**Reply by simonxy** [reply]
This solved mine as well, thanks for the tips.

**Comment by jinsl00000** [comment]
the value ‘key’ in RustDeskServer.sh may wrong

**Comment by rustdesk** [comment]
#763
